File: Remove Duplicates from Sorted List - Leetcode 83/Remove Duplicates from Sorted List - Leetcode 83.py
class Solution:
    def deleteDuplicates(self, head: Optional[ListNode]) -> Optional[ListNode]:
        cur = head

        while cur and cur.next:
            if cur.val == cur.next.val:
                cur.next = cur.next.next
            else:
                cur = cur.next
        return head

# Time Complexity: O(n)
# Space Complexity: O(1)

File: Remove Duplicates from Sorted List - Leetcode 83/Remove Duplicates from Sorted List - Leetcode 83.cpp
class Solution {
public:
    ListNode* deleteDuplicates(ListNode* head) {
        ListNode* cur = head;

        while (cur != nullptr && cur->next != nullptr) {
            if (cur->val == cur->next->val) {
                cur->next = cur->next->next;
            } else {
                cur = cur->next;
            }
        }

        return head;
    }
};
